app.py

```diff
@@ -0,0 +1,208 @@
+# AGPL: a notification must be added stating that changes have been made to that file.
+
```

Comment:
don't you want to make it class based? take a look at the following proposal and see if you could get inspired by it 

```

class StreamlitUI:
    def __init__(self):
        # all ui creation done here

class App:
    def __init__(self):
        self.tts = load_model()
        self.ui = StreamlitUI()
        ...
    
    def run(self):
        if not st.button("Start"):
            return
        # if button was clicked, process the input
        # prepare 
        # run inference
            ...


def main():
    app = App()
    app.run()

if __name__ == "__main__":
     main()

Reply:
I'm not really a fan of classes when they don't reduce the amount of code